Created the blog post announcing Data Prepper 2.1.0. #1303

dlvenable opened the pull request:
### Description

The blog post for announcing Data Prepper 2.1.0

Co-authors:

@asifsmohammed
@kkondaka
 
### Issues Resolved

Resolves #1303

### Check List

- [x] Commits are signed per the DCO using --signoff
- [x] Update Krishna's link for authors
- [x] Update Asif's link for authors


By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

pajuric commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,251 @@
+date: 2023-02-07 15:00:00 -0500
+categories:
+  - releases
+---
```
Please add front matter. 

**Meta Keywords**: Data Prepper 2.1.0, Open Telemetry metrics, detect anomalies in the logs and traces, sampling and rate-limiting
**Meta description**: Data Prepper 2.1.0 improves performance and stability with enhancements such as generated metrics, anomaly detection, sampling and limiting, and more.

dlvenable commented:
My co-author @kkondaka submitted a PR with his bio: #1324

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+---
+
+Data Prepper 2.1.0 is now available for [download](https://opensearch.org/downloads.html#data-prepper)! This release
+provides a number of new features to Data Prepper.  Additionally, the maintainers have improved Data Prepper’s 
```
```suggestion
provides several new features to Data Prepper.  Additionally, the maintainers have improved Data Prepper’s 
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+## Get started
+
+You can download Data Prepper or install a Docker container from the OpenSearch downloads page. The maintainers encourage 
```
```suggestion
You can [download](https://opensearch.org/downloads.html) Data Prepper or install a Docker container from the OpenSearch downloads page. The maintainers encourage 
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
````diff
@@ -0,0 +1,253 @@
+         index: sampled-traces
+```
+
+## OTel logs
````
```suggestion
## Otel logs
```

dlvenable replied:
This is an abbreviation for OpenTelemetry which is a proper noun.

> OpenTelemetry, also known as OTel for short,

https://opentelemetry.io/docs/

carolxob replied:
In that case, this should read as: "OpenTelemetry (OTel) logs".

kolchfa-aws replied:
Since we're not using the abbreviation in the section body, does it make sense to leave it in the spelled-out form? "OpenTelemetry logs"?

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+One of Data Prepper’s goals is supporting open standards. Data Prepper now supports the [OpenTelemetry](https://opentelemetry.io)
+log format. Previously, Data Prepper supported log data through the 
+[HTTP source plugin](https://opensearch.org/docs/latest/data-prepper/configuration/sources/http-source/) 
+which works well with tools such as [FluentBit](https://fluentbit.io). Some users would like to deploy the 
```
```suggestion
, which works well with tools such as [FluentBit](https://fluentbit.io). Some users would like to deploy the 
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+Running Data Prepper as an ingestion pipeline in front of OpenSearch is an important capability for Data Prepper. This 
+release now adds hew enhancements to Data Prepper’s OpenSearch sink to make it even more useful with OpenSearch.
+
+Data Prepper now supports routing traces and logs to different indices dynamically. Pipeline authors can define an 
```
```suggestion
Data Prepper now supports routing traces and logs to different indexes dynamically. Pipeline authors can define an 
```

kolchfa-aws replied:
```suggestion
Data Prepper now supports routing traces and logs to different indexes dynamically. To achieve this, you can define an 
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+Data Prepper now supports routing traces and logs to different indices dynamically. Pipeline authors can define an 
+index name using a format string which can include properties from different events. With this capability, Data Prepper 
+can support an arbitrary number of indices using a single sink. For example, if you have logs with different 
+application identifiers, you could route events to an index specifically for each application.
```
```suggestion
application identifiers, you can route events to an index specifically for each application.
```

pajuric commented:
Just confirming this has gone through editorial review by @natebower.

natebower commented:
> Just confirming this has gone through editorial review by @natebower.

@pajuric No, we're still in progress, and the dates for this will likely be adjusted as well.

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+Data Prepper also now allows routing documents to specific OpenSearch shards by use of the existing `routing` parameter 
+in OpenSearch. The new `routing_field` property on the OpenSearch sink will use properties from events to specify how to 
+route within OpenSearch. Most users will prefer to allow OpenSearch to generate document ids and choose the shard 
```
```suggestion
route within OpenSearch. Most users will prefer to allow OpenSearch to generate document IDs and choose the shard 
```

kolchfa-aws replied:
```suggestion
route them within OpenSearch. Most users prefer allowing OpenSearch to generate document IDs and choose the shard 
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+from that. But, in some cases users need to specify these values and now Data Prepper offers users that option.
+
+Going hand-in-hand with the routing field, Data Prepper now allows pipeline authors to specify a complex field to 
+use when specifying a document Id. Previously, a pipeline author would need to copy a value to a field in the root of 
```
```suggestion
use when specifying a document ID. Previously, a pipeline author would need to copy a value to a field in the root of 
```

kolchfa-aws replied:
```suggestion
use when specifying a document ID. Previously you could only copy a value to a field in the root of 
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+Going hand-in-hand with the routing field, Data Prepper now allows pipeline authors to specify a complex field to 
+use when specifying a document Id. Previously, a pipeline author would need to copy a value to a field in the root of 
+the event to use it as a document Id.
```
```suggestion
the event to use it as a document ID.
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+Anomaly detection in Data Prepper allows for more scalable and efficient way to detect anomalies in the logs and traces 
+because the ML algorithm is applied on the server side.
+
+For example, the anomaly detector processor placed after aggregate processor that generates histogram of trace latencies 
```
```suggestion
For example, the anomaly detector processor placed after aggregate processor that generates a histogram of trace latencies 
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+## Sampling and limiting
+
+Data Prepper 2.1 supports sampling and rate-limiting to limit the number of events that are sent to a Sink.
```
```suggestion
Data Prepper 2.1 supports sampling and rate-limiting to limit the number of events that are sent to a sink.
```

kolchfa-aws replied:
```suggestion
Data Prepper 2.1 supports sampling and rate limiting to limit the number of events that are sent to a sink.
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+Data Prepper 2.1 supports sampling and rate-limiting to limit the number of events that are sent to a Sink.
+
+One possible use case for both these features can be used to reduce the load on the OpenSearch when storing normal 
```
```suggestion
One possible use case for both these features is to reduce the load on the OpenSearch when storing normal 
```

kolchfa-aws replied:
Could we say "ordinary" instead of "normal"?

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+Both these features are available as a configurable action in the aggregate processor. Simply choose the appropriate 
+action (sampling or rate limiting) and specify the sampling percent or events per second option as needed.
+
+For example, the following percent sampler configuration sends 60% of the events to OpenSearch
```
```suggestion
For example, the following percent sampler configuration sends 60% of the events to OpenSearch:
```

carolxob commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+excess events. The `block` is useful if there is a temporary burst and the user do not want to lose any events in that 
+case. The default value for this option is `block`.
+
+And the following rate limiter configuration send 10 events per second to the open search and using `drop` option.
```
```suggestion
And the following rate limiter configuration send 10 events per second to the open search and using `drop` option:
```

kolchfa-aws replied:
```suggestion
The following rate-limiting configuration sends 10 events per second to OpenSearch and uses the `drop` option:
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+because the ML algorithm is applied on the server side.
+
+For example, the anomaly detector processor placed after aggregate processor that generates histogram of trace latencies 
+(`durationInNanos`) could detect abnormally high latencies. The following anomaly detector processor configuration 
```
```suggestion
The following is an example of such an anomaly detector processor configuration: 
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+For example, the anomaly detector processor placed after aggregate processor that generates histogram of trace latencies 
+(`durationInNanos`) could detect abnormally high latencies. The following anomaly detector processor configuration 
+can be used for it.
```
```suggestion

```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
````diff
@@ -0,0 +1,253 @@
+         index: high-latency-traces-with-anomalies
+```
+
+## Sampling and limiting
````
```suggestion
## Sampling and rate limiting
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+One possible use case for both these features can be used to reduce the load on the OpenSearch when storing normal 
+(not very interesting) logs and metrics.
+
+Both these features are available as a configurable action in the aggregate processor. Simply choose the appropriate 
```
```suggestion
Both sampling and rate limiting are available as configurable actions in the aggregate processor. To use the features, choose the appropriate 
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+(not very interesting) logs and metrics.
+
+Both these features are available as a configurable action in the aggregate processor. Simply choose the appropriate 
+action (sampling or rate limiting) and specify the sampling percent or events per second option as needed.
```
```suggestion
action (sampling or rate limiting) and specify the percentage of events or the number of events per second to send to OpenSearch.
```

kolchfa-aws replied:
The sample percent is for sampling, and the number of events is for rate limiting, correct? If this is the case, I would split the sentence into two parts, each describing one option.

dlvenable replied:
Correct, I will split this.

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
````diff
@@ -0,0 +1,253 @@
+         index: sampled-traces
+```
+
+Rate limiter action can be configured to limit the number of events by specifying the number of events per second and 
````
`A rate-limiting action`?

dlvenable replied:
This sounds reasonable to me.

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
````diff
@@ -0,0 +1,253 @@
+         index: sampled-traces
+```
+
+Rate limiter action can be configured to limit the number of events by specifying the number of events per second and 
````
```suggestion
You can configure a rate-limiting action to limit the number of events by specifying the number of events per second. 
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
````diff
@@ -0,0 +1,253 @@
+```
+
+Rate limiter action can be configured to limit the number of events by specifying the number of events per second and 
+optionally what to do when the number of events exceed the specified limit using the `when_exceeds` option. This 
````
```suggestion
Optionally, you can specify (to Data Prepper?) what to do when the number of events exceeds the specified limit using the `when_exceeds` option. The
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+Rate limiter action can be configured to limit the number of events by specifying the number of events per second and 
+optionally what to do when the number of events exceed the specified limit using the `when_exceeds` option. This 
+`when_exceeds` option can take either `block` or `drop` value and can either block until events are allowed or drop any 
```
```suggestion
`when_exceeds` option can take the following values:
- `block`: block (Data Prepper?) until events are allowed (this is the default value)
- `drop`: drop any excess events
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+Rate limiter action can be configured to limit the number of events by specifying the number of events per second and 
+optionally what to do when the number of events exceed the specified limit using the `when_exceeds` option. This 
+`when_exceeds` option can take either `block` or `drop` value and can either block until events are allowed or drop any 
+excess events. The `block` is useful if there is a temporary burst and the user do not want to lose any events in that 
```
```suggestion
Setting `when_exceeds` to `block` is useful if there is a temporary burst in the number of events and you don't want to lose any events.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+One of Data Prepper’s goals is supporting open standards. Data Prepper now supports the [OpenTelemetry](https://opentelemetry.io)
+log format. Previously, Data Prepper supported log data through the 
+[HTTP source plugin](https://opensearch.org/docs/latest/data-prepper/configuration/sources/http-source/), 
+which works well with tools such as [FluentBit](https://fluentbit.io). Some users would like to deploy the 
```
```suggestion
which works well with tools such as [FluentBit](https://fluentbit.io). If you would like to deploy the 
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+log format. Previously, Data Prepper supported log data through the 
+[HTTP source plugin](https://opensearch.org/docs/latest/data-prepper/configuration/sources/http-source/), 
+which works well with tools such as [FluentBit](https://fluentbit.io). Some users would like to deploy the 
+[OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) on their applications 
```
```suggestion
[OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) on your applications 
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+[HTTP source plugin](https://opensearch.org/docs/latest/data-prepper/configuration/sources/http-source/), 
+which works well with tools such as [FluentBit](https://fluentbit.io). Some users would like to deploy the 
+[OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) on their applications 
+and not require any other sidecars. Those users can now start to send logs to Data Prepper and then to OpenSearch.
```
```suggestion
and not require any other sidecars, you can now send logs to Data Prepper, which will route them to OpenSearch.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+## OpenSearch sink improvements
+
+Running Data Prepper as an ingestion pipeline in front of OpenSearch is an important capability for Data Prepper. This 
+release now adds hew enhancements to Data Prepper’s OpenSearch sink to make it even more useful with OpenSearch.
```
```suggestion
release adds hew enhancements to Data Prepper’s OpenSearch sink to make it even more useful with OpenSearch.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+Data Prepper now supports routing traces and logs to different indices dynamically. Pipeline authors can define an 
+index name using a format string which can include properties from different events. With this capability, Data Prepper 
+can support an arbitrary number of indices using a single sink. For example, if you have logs with different 
```
```suggestion
can support an arbitrary number of indexes using a single sink. For example, if you have logs with different 
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+can support an arbitrary number of indices using a single sink. For example, if you have logs with different 
+application identifiers, you could route events to an index specifically for each application.
+
+Data Prepper also now allows routing documents to specific OpenSearch shards by use of the existing `routing` parameter 
```
```suggestion
Data Prepper now also allows routing documents to specific OpenSearch shards using the existing `routing` parameter 
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+Data Prepper also now allows routing documents to specific OpenSearch shards by use of the existing `routing` parameter 
+in OpenSearch. The new `routing_field` property on the OpenSearch sink will use properties from events to specify how to 
+route within OpenSearch. Most users will prefer to allow OpenSearch to generate document ids and choose the shard 
+from that. But, in some cases users need to specify these values and now Data Prepper offers users that option.
```
What exactly is meant by "choose the shard from that"? Could we add a noun here or explain a bit more?
```suggestion
from that. However, Data Prepper now allows you to specify these values explicitly.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+route within OpenSearch. Most users will prefer to allow OpenSearch to generate document ids and choose the shard 
+from that. But, in some cases users need to specify these values and now Data Prepper offers users that option.
+
+Going hand-in-hand with the routing field, Data Prepper now allows pipeline authors to specify a complex field to 
```
```suggestion
Going hand-in-hand with the routing field, Data Prepper now lets you specify a complex field to 
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+[convert entry](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/mutate-event-processors#convertentryprocessor) 
+processor as part of 
+[mutate processors](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/mutate-event-processors#mutate-event-processors) 
+to convert value from one type to another. This can be particularly useful in conditional expressions when using conditional routing.
```
```suggestion
to convert a value from one type to another. This can be particularly useful in conditional expressions when using conditional routing.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+[mutate processors](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/mutate-event-processors#mutate-event-processors) 
+to convert value from one type to another. This can be particularly useful in conditional expressions when using conditional routing.
+
+For example, in the following pipeline using this processor would allow a user to convert value of `status` key to `integer`. 
```
```suggestion
For example, using the following processor in a pipeline lets you convert the value of `status` key to `integer`:
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+## Other improvements
+
+In addition to the new features already described, Data Prepper 2.1.0 has a few other improvements. We want to highlight a few of them.
```
```suggestion
In addition to the new features already described, Data Prepper 2.1.0 has several other important improvements:
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+In addition to the new features already described, Data Prepper 2.1.0 has a few other improvements. We want to highlight a few of them.
+
+* Data Prepper pipelines with OpenSearch sink now waits for OpenSearch to start during pipeline initialization if the sink is not available instead of failing.
```
```suggestion
* Data Prepper pipelines with OpenSearch sinks now wait for OpenSearch to start during pipeline initialization if the sink is not available instead of failing.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+In addition to the new features already described, Data Prepper 2.1.0 has a few other improvements. We want to highlight a few of them.
+
+* Data Prepper pipelines with OpenSearch sink now waits for OpenSearch to start during pipeline initialization if the sink is not available instead of failing.
+* OpenSearch sink now allows users to load index mapping and ISM template files from S3 bucket. Pipeline authors can configure it by providing S3 URI’s in existing configuration option to load them.
```
So, one S3 bucket can have many URIs? Also, what is the existing configuration option: could we be more specific?
```suggestion
* OpenSearch sink now allows you to load index mapping and ISM template files from an S3 bucket. You can configure the S3 bucket by providing its URIs in existing configuration option to load them.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+
+* Data Prepper pipelines with OpenSearch sink now waits for OpenSearch to start during pipeline initialization if the sink is not available instead of failing.
+* OpenSearch sink now allows users to load index mapping and ISM template files from S3 bucket. Pipeline authors can configure it by providing S3 URI’s in existing configuration option to load them.
+* Data Prepper expressions now support `null` value keyword in conditions with equality operators.
```
```suggestion
* Data Prepper expressions now support the `null` value keyword in conditions with equality operators.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+* Data Prepper pipelines with OpenSearch sink now waits for OpenSearch to start during pipeline initialization if the sink is not available instead of failing.
+* OpenSearch sink now allows users to load index mapping and ISM template files from S3 bucket. Pipeline authors can configure it by providing S3 URI’s in existing configuration option to load them.
+* Data Prepper expressions now support `null` value keyword in conditions with equality operators.
+* To help with pipeline stability, Data Prepper now supports a circuit breaker on Java heap usage. This will cause the sources to stop accepting new data that could possibly cause the process to crash.
```
```suggestion
* To help with pipeline stability, Data Prepper now supports a circuit breaker on Java heap usage. The circuit breaker will cause the sources to stop accepting new data that could possibly cause the process to crash.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+## Get started
+
+You can [download](https://opensearch.org/downloads.html) Data Prepper or install a Docker container from the OpenSearch downloads page. The maintainers encourage 
+all users to update to Data Prepper 2.1.0 to gain the improved stability, administration options, and feature set.
```
```suggestion
all users to update to Data Prepper 2.1.0 to gain improved stability, additional administration options, and new features.
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+You can [download](https://opensearch.org/downloads.html) Data Prepper or install a Docker container from the OpenSearch downloads page. The maintainers encourage 
+all users to update to Data Prepper 2.1.0 to gain the improved stability, administration options, and feature set.
+
+Work is already started on Data Prepper 2.2. Please see the [roadmap](https://github.com/opensearch-project/data-prepper/projects/1)
```
```suggestion
We already started working on Data Prepper 2.2. See the [roadmap](https://github.com/opensearch-project/data-prepper/projects/1)
```

kolchfa-aws commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+all users to update to Data Prepper 2.1.0 to gain the improved stability, administration options, and feature set.
+
+Work is already started on Data Prepper 2.2. Please see the [roadmap](https://github.com/opensearch-project/data-prepper/projects/1)
+to learn about what is upcoming.
```
```suggestion
to learn more.
```

kolchfa-aws reviewed (commented):
Thank you for writing this comprehensive intro blog! It would be easier if this post did not have line breaks so that we could edit complete sentences. Please see my comments and let me know if you'd like to discuss.

natebower commented:
@dlvenable @pajuric Can you coordinate on new dates for this? I know there are some release scheduling considerations to be taken into account.

kkondaka commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,253 @@
+layout: post
+title:  "Announcing Data Prepper 2.1.0"
+authors:
+- kkondaka
```
authors files for first two authors are already created.

dlvenable commented:
@natebower , I have pushed a change addressing the feedback provided thus far.

I think we are looking to publish next week, but should have more information tomorrow.

natebower commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+In addition to the new features already described, Data Prepper 2.1.0 has several other important improvements:
+
+* Data Prepper pipelines with OpenSearch sinks now wait for OpenSearch to start during pipeline initialization if the sink is not available instead of failing.
+*  OpenSearch sink now allows you to load index mapping and ISM template files from an S3 bucket. You can configure the S3 object by providing its S3 URI in existing configuration option to load them.
```
```suggestion
*  OpenSearch sinks now allow you to load index mapping and Index State Management (ISM) template files from an Amazon Simple Storage Service (Amazon S3) bucket. You can configure the S3 object by providing its S3 URI in existing configuration option to load them.
```

natebower replied:
The last sentence is a bit confusingly written, is missing articles, and has an unclear subject. Please revise for simplicity/clarity.

dlvenable replied:
I'll just remove it. I don't think we need it in the release blog post.

natebower commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+* Data Prepper pipelines with OpenSearch sinks now wait for OpenSearch to start during pipeline initialization if the sink is not available instead of failing.
+*  OpenSearch sink now allows you to load index mapping and ISM template files from an S3 bucket. You can configure the S3 object by providing its S3 URI in existing configuration option to load them.
+* Data Prepper expressions now support the `null` value keyword in conditions with equality operators.
+* To help with pipeline stability, Data Prepper now supports a circuit breaker on Java heap usage. The circuit breaker will cause the sources to stop accepting new data that could possibly cause the process to crash.
```
```suggestion
* To help with pipeline stability, Data Prepper now provides a circuit breaker on Java heap usage. The circuit breaker will cause the sources to stop accepting new data that could possibly cause the process to crash.
```

natebower commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+*  OpenSearch sink now allows you to load index mapping and ISM template files from an S3 bucket. You can configure the S3 object by providing its S3 URI in existing configuration option to load them.
+* Data Prepper expressions now support the `null` value keyword in conditions with equality operators.
+* To help with pipeline stability, Data Prepper now supports a circuit breaker on Java heap usage. The circuit breaker will cause the sources to stop accepting new data that could possibly cause the process to crash.
+* The documentation team has been updating the [Data Prepper documentation](https://opensearch.org/docs/latest/data-prepper/index/) by migrating many of the details from the GitHub repository into the primary documentation. 
```
```suggestion
* The Documentation team has been updating the [Data Prepper documentation](https://opensearch.org/docs/latest/data-prepper/index/) by migrating many of the details from the GitHub repository to the primary documentation. 
```

natebower replied:
Is this sentence necessary? We should probably avoid referring to the Documentation team as a discrete entity, given that this is open source, and I'm not sure how much value this sentence adds.

dlvenable replied:
We can remove it. I was calling it out because the documentation is much better than it used to be.

natebower commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+
+See the [release notes](https://github.com/opensearch-project/data-prepper/releases/tag/2.1.0) for a full list of changes.
+
+## Get started
```
```suggestion
## Getting started
```

natebower commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+## Get started
+
+You can [download](https://opensearch.org/downloads.html) Data Prepper or install a Docker container from the OpenSearch downloads page. The maintainers encourage 
+all users to update to Data Prepper 2.1.0 to gain the improved stability, additional administration options, and new features.
```
```suggestion
all users to update to Data Prepper 2.1.0 in order to gain the improved stability, additional administrative options, and new features.
```

natebower reviewed (changes requested):
@dlvenable Please see my comments and changes and let me know if you have any questions. Tag me once you've had a chance to address my comments, and I'll take another look and approve when ready. Thanks!

dlvenable commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+
+## Get started
+
+You can [download](https://opensearch.org/downloads.html) Data Prepper or install a Docker container from the OpenSearch downloads page. The maintainers encourage 
```
I'm not sure that "on" is the right preposition here. The verbs here are "download" and "install." Users are installing on their local machine from OpenSearch.

natebower replied:
Ah ok, I see the distinction. In that case, you're good with "from".

dlvenable commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+(not very interesting) logs and metrics.
+
+Both sampling and rate limiting are available as configurable actions in the aggregate processor.
+To use the features, choose the appropriate action (sampling or rate limiting).
```
I think that is a good idea.

dlvenable commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+Data Prepper now supports routing traces and logs to different indexes dynamically. To achieve this, you can define an 
+index name using a format string which can include properties from different events. With this capability, Data Prepper 
+can support an arbitrary number of indexes using a single sink. For example, if you have logs with different 
+application identifiers, you can route events to an index specifically for each application.
```
I like your suggestion. Would "unique" help here?

> to an index unique to each application

natebower replied:
Yes, that would work :-)

pajuric commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+date: 2023-02-07 15:00:00 -0500
+categories:
+  - releases
+meta_keywords: Data Prepper 2.1.0, Open Telemetry metrics, OpenTelemetry logs, detect anomalies in logs and traces, sampling and rate-limiting
```
Please update the post date for the blog to March 2, 2023.

dlvenable replied:
@natebower , I made this change, but I also see that the original comment for this said not to edit:

https://github.com/opensearch-project/project-website/pull/1314#discussion_r1095172811

Should I change or not?

pajuric replied:
@dlvenable you can make the changes. Technically, browsers just check that the words are strung together and do not care about the spacing, punctuation, or capitalization.

natebower replied:
@dlvenable Thanks! @pajuric I'll definitely be editing these descriptions if I see errors in them.

dlvenable commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+
+You can combine both of these actions with aggregate processor’s conditional aggregation option to generate more meaningful metrics.
+
+For example, you can obtain the number of traces with error (`status_code` equal to 2) using the `count` aggregate action along 
```
@kkondaka , Can you help with rewording this part?

kkondaka replied:
@natebower, what would be the correct way of writing this? I meant that the traces with status code 2 are considered error. Something like this is more clear? - "the number of traces with error, ie with `status_code` equals to 2. 
Basically traces have a field called `status_code` and if the `status_code` value is 2, it is considered as error. That's what I meant by "traces with error". Appreciate your feedback on how to change the wording to reflect the correct meaning.

natebower replied:
It looks like it just needed an article (an).

kkondaka replied:
@natebower do you mean I just put "an" before "error" - like - " the number of trace with an error (`status_code` equal to 2) using the `count` aggregate action..."

natebower replied:
@kkondaka Correct, I've already made the change in a suggestion during my last review :-)

kkondaka replied:
@natebower, great! Thanks!

dlvenable commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,256 @@
+Data Prepper 2.1 supports sampling and rate-limiting to limit the number of events that are sent to a sink.
+
+One possible use case for both these features is to reduce the load on the OpenSearch when storing ordinary 
+(not very interesting) logs and metrics.
```
I gave it another shot. Please see the change.

dlvenable commented:
@natebower , I have addressed most of the PR comments. I left you a few questions and have one more item to reword.

natebower commented:
> @natebower , I have addressed most of the PR comments. I left you a few questions and have one more item to reword.

@dlvenable Thanks! I'll take another look today.

natebower commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,254 @@
+
+You can combine both of these actions with the aggregate processor’s conditional aggregation option in order to generate more meaningful metrics.
+
+For example, you can obtain the number of traces with error (`status_code` equal to 2) using the `count` aggregate action along 
```
```suggestion
For example, you can obtain the number of traces with an error (`status_code` equal to 2) using the `count` aggregate action along 
```

natebower commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,254 @@
+
+Data Prepper 2.1 supports sampling and rate limiting to limit the number of events that are sent to a sink.
+
+These features can be used to reduce the load on the OpenSearch cluster when storing uninteresting logs and metrics.
```
```suggestion
These features can be used to reduce the load on the OpenSearch cluster when storing ordinary logs and metrics.
```

dlvenable replied:
@natebower , Is there a specific reason you recommended going back to "ordinary" here? It does seem that "uninteresting" is a more apt word because these are logs that somebody would not be very interested in.

natebower replied:
@dlvenable I think "ordinary" is ok as long as we qualify what we're talking about, which I believe you did in the sentence that follows this one. "Uninteresting" is subjective and a bit awkward in this context.

natebower commented on _posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md:
```diff
@@ -0,0 +1,254 @@
+Data Prepper 2.1 supports sampling and rate limiting to limit the number of events that are sent to a sink.
+
+These features can be used to reduce the load on the OpenSearch cluster when storing uninteresting logs and metrics.
+For example, you can limit the number of success HTTP logs or repetitive requests such as from a health check. 
```
```suggestion
For example, you can limit the number of success HTTP logs or repetitive requests, such as from a health check. 
```

natebower reviewed (commented):
@dlvenable I believe I answered all your questions. Let me know when you're finished rewording, and I'll approve. Thanks!

natebower reviewed (approved):
LGTM. Approved and ready to publish. Thanks, David!

pajuric commented:
@krisfreedain - This is ready to publish on 3/2/23

krisfreedain commented:
@dlvenable are the release notes on their way out? the link for them (https://github.com/opensearch-project/data-prepper/releases/tag/2.1.0) is still 404

dlvenable commented:
@krisfreedain , We will push the release notes in roughly an hour.

peterzhuamazon commented:
> @krisfreedain , We will push the release notes in roughly an hour.

Hi @krisfreedain @dlvenable ,

I am releasing the 2.1.0 version with @asifsmohammed now.

Thanks.
